File: Backend/entity/roomlayout.go
package entity
import (
	"gorm.io/gorm"
)

type RoomLayout struct{
	gorm.Model
	BuildingID uint
	Building Building `json:"BuildingID"`

	RoomID uint
	Room   *Room `gorm:"foreignKey:RoomID"`

	FloorID uint
	Floor Floor `json:"foreignKey:FloorID"`

	PositionX int `json:"position_x"`

	PositionY int `json:"position_y"`
}

File: Frontend/src/interfaces/RoomLayout.ts
export interface RoomLayoutInterface {
    room_layout_id: number;  
    room_number: string;
    room_name: string;
    position_x: number;
    position_y: number;
}


// กำหนดประเภทของ roomLayout เป็นอาร์เรย์ 2 มิติของ Room
export type RoomLayout = RoomLayoutInterface[][];
